docs/nachet-model-documentation.md

```diff
@@ -0,0 +1,191 @@
+# Nachet Interactive Models
+
+## Executive Summary
+
+Nachet Interactive uses various models to detect seeds. Documentation is
+essential to keep track of their features. The models can perform different
+tasks, including Image Classification, Image Segmentation, and Object Detection.
+
+## Task
+
+Nachet Interactive models' perform the following tasks:
+
+|Task|Action|Input/Output|
+|:--|:--|:-----|
+|[Classification](https://huggingface.co/tasks/image-classification) | This task involves assigning a single label or class to each image the model receives. | The input for the classification models is an image, and the output is a prediction of the class it belongs to. |
+|[Object Detection](https://huggingface.co/tasks/object-detection) | Identify and locate an object belonging to a specific class within an image. |The object detection models take an image as an input and output the image with a label and a box around the detected object. |
+|[Segmentation](https://huggingface.co/tasks/image-segmentation) | Segmentation is the task of dividing images into different parts, where each pixel in the image is mapped to an object. It includes instance segmentation, panoptic segmentation, and semantic segmentation.| The segmentation models take an image as input and return an image divided into objects. |
+
+> As of today (2024-02-22), no model uses segmentation. To know more about each
+> task, click on the task to follow the link to their hugging face page. To know
+> more about AI tasks in general: [Hugging Face
+> Tasks](https://huggingface.co/tasks)
+
+## List of models
+
+|Model|Full name|Task|API Call Function|Inference Function|Active|Accuracy|
+|--|--|:--:|:--:|:--:|:--:|:--:|
+|Nachet-6seeds | m-14of15seeds-6seedsmag | Object Detection | nachet_6seeds | None | Yes | - |
+|Seed-detector | seed-detector-1 |  Object Detection | seed_detector | process_image_slicing | Yes | - |
+|Swin | swinv1-base-dataaugv2-1 | Classification | swin | process_swin_result | Yes | - |
+
+### Inference and API Call Function
+
+The inference and API call functions act as entry and exit points for the model. The API call explicitly requests a prediction from the specified model (such as Swin, Nachet-6seeds, etc.). The inference function processes the data before sending it to the frontend if the model requires it. For instance, the Seed-detector only returns "seed" as a label, and its inference needs to be processed and passed to the next model which assigns the correct label to the seeds.
+
+## Return value of models
+
+```json
+{
+    "filename": "tmp/tmp_file_name",
+    "boxes": [
+        {"box": {
+                "topX": 0.0,
+                "topY": 0.0,
+                "bottomX": 0.0,
+                "bottomY": 0.0
```

Comment:
I'm not sure to understand what is topX, topY, bottomX, bottomY. Is it the equivalent of saying left, upper, right, lower ?

Reply:
It's the coordinates used by the canvas in the frontend to draw the red boxes around the seed. topX and topY are the top-left coordinates while bottomX and bottomY are the bottom-right coordinates.

Reply:
Adding a basic picture in `nachet-model-documentation.md` to show the box and its coordinates could help people understand better.

Moreover, it might be advantageous to adopt the terminology "left, upper, right, lower" for this context. These terms align with the conventions utilized in the [Pillow library](https://pillow.readthedocs.io/en/stable/reference/Image.html) for defining bounding boxes, thereby offering familiarity and consistency to users. For instance, "topX" corresponds to "upper" and "topY" to "left,"

Reply:
Good point! I will add the image to the documentation. As for the terminology, I'm a bit reluctant to change it to reflect Pillow usage since the keys are designed to fit the frontend convention.  [Here's out it is us in the frontend](https://github.com/ai-cfia/nachet-frontend/blob/c19a1c236f39894ca9b0be2ffe1856fe00fcc7cb/src/root/body/index.tsx#L542)